apps/crud/src/app/app.component.ts

```diff
@@ -1,52 +1,153 @@
+
+type IodoItem = {
+  todo: Todo;
+  loading: boolean;
+  error: Error | null;
+};
 
 @Component({
+  selector: 'todo-item',
+  imports: [CommonModule, LetModule, MatProgressSpinnerModule, MatButtonModule],
   standalone: true,
-  imports: [CommonModule],
-  selector: 'app-root',
   template: `
-    <div *ngFor="let todo of todos">
-      {{ todo.title }}
-      <button (click)="update(todo)">Update</button>
-    </div>
+    <ng-container *ngIf="vm$ | async as vm">
+      <p *ngIf="vm?.error">{{ vm?.error?.message }}</p>
+
+      <mat-spinner *ngIf="vm?.loading" [diameter]="40"> </mat-spinner>
+
+      <ng-container *ngIf="!vm?.error && !vm?.loading">
+        <div *ngrxLet="vm.todo as todo">
+          {{ todo.title }}
+          <button
+            mat-stroked-button
+            color="primary"
+            (click)="updateTodo$(todo)">
+            Update
+          </button>
+          <button
+            mat-stroked-button
+            color="warn"
+            (click)="todoService.deleteTodo$(todo)">
+            Delete
+          </button>
+        </div>
+      </ng-container>
+    </ng-container>
   `,
-  styles: [],
+  providers: [ComponentStore],
+  styles: [
+    `
+      :host {
+        padding: 20px;
+        display: flex;
+        gap: 10px;
+      }
+    `,
+  ],
 })
-export class AppComponent implements OnInit {
-  todos!: any[];
-
-  constructor(private http: HttpClient) {}
-
-  ngOnInit(): void {
-    this.http
-      .get<any[]>('https://jsonplaceholder.typicode.com/todos')
-      .subscribe((todos) => {
-        console.log('return', todos);
-        this.todos = todos;
-      });
+export class TodoItemComponent extends ComponentStore<IodoItem> {
+  readonly vm$ = this.select({
+    todo: this.select((state) => state.todo),
+    loading: this.select((state) => state.loading),
+    error: this.select((state) => state.error),
+  });
+
+  constructor(public todoService: TodoService) {
+    super({
+      todo: {} as Todo,
+      loading: false,
+      error: null,
+    });
   }
 
-  update(todo: any) {
-    this.http
-      .put<any>(
-        `https://jsonplaceholder.typicode.com/todos/${todo.id}`,
-        JSON.stringify({
-          todo: todo.id,
-          title: randText(),
-          body: todo.body,
-          userId: todo.userId,
-        }),
-        {
-          headers: {
-            'Content-type': 'application/json; charset=UTF-8',
-          },
+  @Input()
+  set todoItem(todo: Todo) {
+    this.patchState({
+      todo,
+      loading: false,
+      error: null,
+    });
+  }
+
+  readonly updateTodo$ = this.effect((todo: Observable<Todo>) => {
+    return todo.pipe(
+      tap(() => this.showLoading()),
+      switchMap((todo) => this.todoService.updateTodo(todo)),
+      tapResponse(
+        (todo) => {
+          this.setState({
+            todo,
+            loading: false,
+            error: null,
+          });
+        },
+        (error: Error) => {
+          this.handleError(error);
```

Comment:
warning: updateTodo is updating TodoItem inside this component but not inside the main todoList, your two items are not synchronised anymore. This can be error-prone.

Reply:
okay, forget that, thank you @tomalaforge